lib/versionmanager.js

```diff
@@ -441,6 +458,37 @@ async function queryVersions(packageMap, options = {}) {
   return _.pickBy(zipVersions(versions), _.identity)
 }
 
+/**
+ * Get the latest or greatest versions from the NPM repository based on the version target.
+ *
+ * @param packageMap   An object whose keys are package name and values are version
+ * @param [options={}] Options.
+ * @returns Promised {packageName: peer dependencies} collection
+ */
+async function getPeerDependenciesFromRegistry(packageMap, options) {
+  const numItems = Object.keys(packageMap).length
+  let bar
+  if (!options.json && options.loglevel !== 'silent' && options.loglevel !== 'verbose' && numItems > 0) {
+    bar = new ProgressBar('[:bar] :current/:total :percent', { total: numItems, width: 20 })
+    bar.render()
+  }
+
+  const packageManager = getPackageManager(options.packageManager)
+  let result = {}
+  if (packageManager.getPeerDependencies) {
```

Comment:
Rather than use an if statement here, it would be better to do the opposite check and short circuit the logic:

```js
if (!packageManager.getPeerDependencies) return {}
```

This eliminates the mutation and results in flatter code. You can put the conditional above the `ProgressBar` instantiation.